From c3eb00ba7a4e5316b066b5c3a39c35e1db9b0be7 Mon Sep 17 00:00:00 2001
From: Sunil Goutham <sgoutham@marvell.com>
Date: Mon, 12 Aug 2019 17:18:55 +0530
Subject: [PATCH 314/767] octeontx2-pf: Fix VF id in the FLR handler

commit 1ffaf44fda1307be64755ec8ccf030da724b33ea from
git@git.assembla.com:cavium/WindRiver.linux.git

A VF id of 64 in the FLR handler will result
in BIT_ULL(64) which is useless.

Change-Id: I4ff6644342829c5382e8b92123b1e4e73895a408
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/14393
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
index 60b368539c3b..392d83152d2b 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
@@ -100,7 +100,7 @@ static void otx2_flr_handler(struct work_struct *work)
 	req->hdr.pcifunc |= (vf + 1) & RVU_PFVF_FUNC_MASK;
 
 	if (!otx2_sync_mbox_msg(&pf->mbox)) {
-		if (vf > 64) {
+		if (vf >= 64) {
 			reg = 1;
 			vf = vf - 64;
 		}
-- 
2.31.1

